build.gradle

```diff
@@ -0,0 +1,292 @@
+buildscript {
+  repositories {
+    maven {
+      url 'https://plugins.gradle.org/m2/'
+    }
+  }
+  dependencies {
+    // check updates with https://plugins.gradle.org/
+    classpath 'io.spring.gradle:dependency-management-plugin:1.0.4.RELEASE'
+    classpath 'com.github.jengelman.gradle.plugins:shadow:2.0.2'
+    classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.4'
+    classpath 'com.moowork.gradle:gradle-node-plugin:1.2.0'
+    classpath 'org.jfrog.buildinfo:build-info-extractor-gradle:4.6.2'
+    classpath 'org.sonarsource.scanner.gradle:sonarqube-gradle-plugin:2.6.2'
+  }
+}
+
+// analyze SonarQube with SonarQube!
+apply plugin: 'org.sonarqube'
+sonarqube {
+  properties {
+    property 'sonar.projectName', sonarProjectName
+  }
+}
+
+allprojects {
+  apply plugin: 'com.jfrog.artifactory'
+
+  //Replaces the version defined in sources, usually x.y-SNAPSHOT, by a version identifying the build.
+  def buildNumber = System.getProperty("buildNumber")
+  if (version.endsWith('-SNAPSHOT') && buildNumber != null) {
+    version = version.replace('-SNAPSHOT', ".0.$buildNumber")
+  }
+
+  ext {
+    release = project.hasProperty('release') && project.getProperty('release')
+  }
+
+  repositories {
+    def repository = project.hasProperty('cix') ? 'sonarsource-qa' : 'sonarsource'
+    maven {
+      url "https://repox.sonarsource.com/${repository}"
+      def artifactoryUsername = System.env.'ARTIFACTORY_PRIVATE_USERNAME'
+      def artifactoryPassword = System.env.'ARTIFACTORY_PRIVATE_PASSWORD'
+      if (artifactoryUsername && artifactoryPassword) {
+        credentials {
+          username artifactoryUsername
+          password artifactoryPassword
+        }
+      }
+    }
+  }
+}
+
+subprojects {
+  // do not deploy to Artifactory by default
+  artifactoryPublish.skip = true
+
+  ext {
+    elasticsearchVersion = '5.6.3'
+    protobufVersion = '3.5.1'
+  }
+
+  sonarqube {
+    properties {
+      property 'sonar.moduleKey', project.group + ':' + project.name
+    }
+  }
+
+  apply plugin: "jacoco"
```

Comment:
Order of `apply plugin` can be important (for example protobuf must be after java). So I suggest to move all these commands to the top of `subprojects` section.